hyperion/app/creation_steps/workspace/file_manager.py

```
@@ -0,0 +1,577 @@
+                    "Workspace path is not set in context", file_path=str(path)
+                )
+
+            workspace_path = Path(context.workspace_path).resolve()
+            target_path = Path(path).resolve()
+
+            try:
+                target_path.relative_to(workspace_path)
+            except ValueError:
+                raise FileSystemException(
+                    f"Path is outside workspace boundaries: {path}", file_path=str(path)
+                )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)